compiler/lib/eval.ml

```diff
@@ -119,22 +218,131 @@ let eval_prim x =
       | "caml_ceil_float", _ -> float_unop l ceil
+      | "caml_floor_float", _ -> float_unop l floor
+      | "caml_trunc_float", _ -> float_unop l Float.trunc
+      | "caml_round_float", _ -> float_unop l Float.round
       | "caml_cos_float", _ -> float_unop l cos
       | "caml_exp_float", _ -> float_unop l exp
-      | "caml_floor_float", _ -> float_unop l floor
+      | "caml_exp2_float", _ -> Extra_math_op.exp2 l
+      | "caml_expm1_float", _ -> float_unop l expm1
       | "caml_log_float", _ -> float_unop l log
+      | "caml_log1p_float", _ -> float_unop l log1p
+      | "caml_log2_float", _ -> Extra_math_op.log2 l
+      | "caml_log10_float", _ -> float_unop l log10
+      | "caml_cosh_float", _ -> float_unop l cosh
+      | "caml_sinh_float", _ -> float_unop l sinh
+      | "caml_tanh_float", _ -> float_unop l tanh
+      | "caml_acosh_float", _ -> Extra_math_op.acosh l
+      | "caml_asinh_float", _ -> Extra_math_op.asinh l
+      | "caml_atanh_float", _ -> Extra_math_op.atanh l
       | "caml_power_float", _ -> float_binop l ( ** )
       | "caml_sin_float", _ -> float_unop l sin
       | "caml_sqrt_float", _ -> float_unop l sqrt
+      | "caml_cbrt_float", _ -> Extra_math_op.cbrt l
       | "caml_tan_float", _ -> float_unop l tan
+      | "caml_copysign_float", _ -> float_binop l copysign
+      | "caml_signbit_float", _ -> float_unop_bool l Float.sign_bit
+      | "caml_erf_float", _ -> Extra_math_op.erf l
+      | "caml_erfc_float", _ -> Extra_math_op.erfc l
+      | "caml_nextafter_float", _ -> float_binop l Float.next_after
+      | "caml_float_compare", [ Float i; Float j ] ->
+          Some (Int (Targetint.of_int_exn (Float.compare i j)))
+      | "caml_ldexp_float", [ Float f; Int i ] ->
+          Some (Float (ldexp f (Targetint.to_int_exn i)))
+      (* int32 *)
+      | "caml_int32_bits_of_float", [ Float f ] -> int32 (Int32.bits_of_float f)
+      | "caml_int32_float_of_bits", [ Int i ] ->
+          Some (Float (Int32.float_of_bits (Targetint.to_int32 i)))
+      | "caml_int32_float_of_bits", [ Int32 i ] -> Some (Float (Int32.float_of_bits i))
+      | "caml_int32_of_float", [ Float f ] -> int32 (Int32.of_float f)
+      | "caml_int32_to_float", [ Int32 i ] -> Some (Float (Int32.to_float i))
+      | "caml_int32_neg", _ -> int32_unop l Int32.neg
+      | "caml_int32_add", _ -> int32_binop l Int32.add
+      | "caml_int32_sub", _ -> int32_binop l Int32.sub
+      | "caml_int32_mul", _ -> int32_binop l Int32.mul
+      | "caml_int32_and", _ -> int32_binop l Int32.logand
+      | "caml_int32_or", _ -> int32_binop l Int32.logor
+      | "caml_int32_xor", _ -> int32_binop l Int32.logxor
+      | "caml_int32_div", [ _; Int32 i ] when not (Int32.equal i 0l) ->
+          int32_binop l Int32.div
+      | "caml_int32_mod", [ _; Int32 i ] when not (Int32.equal i 0l) ->
+          int32_binop l Int32.rem
+      | "caml_int32_shift_left", _ -> int32_shiftop l Int32.shift_left
```

Comment:
Many of theses are currently only meaningful to the wasm backend. There reason is that the js backend alias some int32/nativeint primitives to their int counterpart.
I find this state quite confusing so I'm experimenting with #1915.